lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/client/MultiRuntimeConfigurationBuilder.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.multi.client;
+
+import org.apache.commons.lang.Validate;
+import org.apache.reef.runtime.local.client.parameters.MaxNumberOfEvaluators;
+import org.apache.reef.runtime.multi.client.parameters.DefaultRuntimeName;
+import org.apache.reef.runtime.multi.client.parameters.RuntimeNames;
+import org.apache.reef.runtime.yarn.client.ExtensibleYarnClientConfiguration;
+import org.apache.reef.runtime.yarn.driver.RuntimeIdentifier;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.formats.ConfigurationModuleBuilder;
+import org.apache.reef.util.Optional;
+
+import java.util.*;
+
+/**
+ * A builder for Multi Runtime Configuration.
+ */
+public final class MultiRuntimeConfigurationBuilder {
```

Comment:
I don't think it is. Once we will have all runtimes supported as a submission runtime we won't have any restrictions. NOthing prevents having Yarn/HDI runtime for example and I can also enable local as a primary but it doesn't make a lot of sense. In order to enable Mesos we will need to define ExtensibleConfiguration for it and test so it can be done once someone will be interested in it and will be able to test it. We can have a JIRA for enabling MEsos support for multi-runtime.